javascript/i18n/phonenumbers/shortnumberinfo.js

```diff
@@ -0,0 +1,710 @@
+ *     empty string when the metadata does not contain such information.
+ */
+i18n.phonenumbers.ShortNumberInfo.prototype.getExampleShortNumber =
+    function(regionCode) {
+  var phoneMetadata = this.getMetadataForRegion_(regionCode);
+  if (!phoneMetadata) {
+    return '';
+  }
+  var desc = phoneMetadata.getShortCode();
+  if (desc.hasExampleNumber()) {
+    return desc.getExampleNumber();
+  }
+  return '';
+};
+
+
+/**
+ * Gets a valid short number for the specified cost category.
+ *
+ * @param {string} regionCode the region for which an example short number is
+ *     needed
+ * @param {i18n.phonenumbers.ShortNumberInfo.ShortNumberCost} cost the cost
+ *     category of number that is needed
+ * @return {i18n.phonenumbers.PhoneNumber} a valid short number for the
+ *     specified region and cost category. Returns an empty string when the
+ *     metadata does not contain such information, or the cost is UNKNOWN_COST.
+ */
+i18n.phonenumbers.ShortNumberInfo.prototype.getExampleShortNumberForCost =
+    function(regionCode, cost) {
+  var phoneMetadata = this.getMetadataForRegion_(regionCode);
+  if (!phoneMetadata) {
+    return '';
+  }
+  var ShortNumberCost = i18n.phonenumbers.ShortNumberInfo.ShortNumberCost;
+  var desc = null;
+  switch (cost) {
+    case ShortNumberCost.TOLL_FREE:
+      desc = phoneMetadata.getTollFree();
+      break;
+    case ShortNumberCost.STANDARD_RATE:
+      desc = phoneMetadata.getStandardRate();
+      break;
+    case ShortNumberCost.PREMIUM_RATE:
+      desc = phoneMetadata.getPremiumRate();
+      break;
+    default:
+      // UNKNOWN_COST numbers are computed by the process of elimination from
+      // the other cost categories.
+  }
+  if (desc && desc.hasExampleNumber()) {
+    return desc.getExampleNumber();
+  }
+  return '';
+};
+
+
+/**
+   * Returns true if the given number, exactly as dialed, might be used to
+   * connect to an emergency service in the given region.
+   * <p>
+   * This method accepts a string, rather than a PhoneNumber, because it needs
+   * to distinguish cases such as "+1 911" and "911", where the former may not
+   * connect to an emergency service in all cases but the latter would. This
+   * method takes into account cases where the number might contain formatting,
+   * or might have additional digits appended (when it is okay to do that in
+   * the specified region).
+   *
+   * @param {number} number the phone number to test
+   * @param {string} regionCode the region where the phone number is being
+   *     dialed
+   * @return {boolean} whether the number might be used to connect to an
+   *     emergency service in the given region
+   */
+i18n.phonenumbers.ShortNumberInfo.prototype.connectsToEmergencyNumber =
+    function(number, regionCode) {
+  return this.matchesEmergencyNumberHelper_(number, regionCode,
+      true /* allows prefix match */);
+};
+
+
+/**
+   * Returns true if the given number exactly matches an emergency service
```

Comment:
Indentation of comments is of
   * 
instead of
  * 

for the following method too.